### PR TITLE
[IA-1393] fixing memory leak

### PR DESCRIPTION
**Pull request by jdcanas**
Seems to work locally™.

### CODE REVIEW

**Review by rtitle — COMMENTED** [review]
Not sure about having these methods return Futures. If `beforeAll` in AuthProviderSpec doesn’t wait for the server to start then could we have race conditions? E.g. test failures because the server hasn’t started yet.

Maybe we could slap on a `startProxyServer.futureValue` in `AuthProviderSpec.beforeAll` to ensure the server is started before tests start running. Does that make sense?

**Comment by jdcanas** [comment]
> Not sure about having these methods return Futures. If `beforeAll` in AuthProviderSpec doesn’t wait for the server to start then could we have race conditions? E.g. test failures because the server hasn’t started yet.
> 
> Maybe we could slap on a `startProxyServer.futureValue` in `AuthProviderSpec.beforeAll` to ensure the server is started before tests start running. Does that make sense?

I think the change I just made accomplishes this, but its possible it doesn't as the mistake is due to unfamiliarity with futures. I originally thought the .futureValue was extracting from the future in before due to the type requirement of how it was being used in afterAll. 

So lifting the server out of the future with .futureValue ensures the future is complete prior to execution passing the line it is on? Is it an antipattern to await on isCompleted or onComplete, which strikes me as more intuitive? 

If futureValue behaves as we're describing it, I think I can change the afterAll from `Await.result(serverBinding, 10.seconds)
        .terminate(hardDeadline = 3.seconds)` to `serverBinding.terminate(hardDeadline = 3.seconds)` where `serverBinding` is the return of `serverBinding.futureValue`

**Review by rtitle — APPROVED** [review]

**Comment by rtitle** [comment]
Yup `.futureValue` is something from ScalaTest. It waits for the future to complete and adds some extra assertions. See http://doc.scalatest.org/3.0.1-2.12/org/scalatest/concurrent/ScalaFutures.html

I think the current code addresses the problem I mentioned above. 👍

File: http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/TestProxy.scala [hunk 1]
```diff
@@ -52,11 +56,21 @@ trait TestProxy { this: ScalaFutures =>
...
   def shutdownProxyServer() = {
-    serverBinding.unbind().futureValue
+    val onceAllConnectionsTerminated: Future[Http.HttpTerminated] =
+      Await.result(serverBinding, 10.seconds)
+        .terminate(hardDeadline = 3.seconds)
+
+    // once all connections are terminated,
+    // - you can invoke coordinated shutdown to tear down the rest of the system:
+    onceAllConnectionsTerminated.flatMap { _ =>
+      system.terminate()
```
Comment (Qi77Qi) [inline http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/TestProxy.scala:68]: 👍 on terminating `system`...but is `Future` really needed?

Comment (jdcanas) [reply http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/TestProxy.scala:68]: this is how the documentation recommended doing it, not sure the advantage though

**Review by Qi77Qi — APPROVED** [review]

**Comment by Qi77Qi** [comment]
:shipit:  thanks for taking this on!

**Pull request merged**